Assets/Mirror/Runtime/Authentication/Authenticator.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Mirror
+{
+    /// <summary>
+    /// Unity Event for the NetworkConnection
+    /// </summary>
+    [Serializable] public class UnityEventConnection : UnityEvent<NetworkConnection> { }
+
+    /// <summary>
+    /// Abstract class for implementing component-based authentication during the Connect phase
+    /// </summary>
+    public abstract class Authenticator : MonoBehaviour
+    {
+        /// <summary>
+        /// Notify subscribers on the server when a client is authenticated
+        /// </summary>
+        [HideInInspector] public UnityEventConnection OnServerAuthenticated = new UnityEventConnection();
```

Comment:
why HideInInspector? could be useful to assign directly in the inspector instead of with code

Reply:
Good point, the user might also want to subscribe to this

Reply:
Removed